### PR TITLE
embedding_lookup_sparse divide by zero fixed

### LINKED ISSUE #36359: [Bug] embedding_lookup_sparse divide by zero when use mean combiner
I come across this issue with TF==1.14.
I found this bug was fixed in this PR: https://github.com/tensorflow/tensorflow/pull/21757.
However, when I checked the implementations in 1.13, 1.14 and 1.15 I found all of them implemented as following:

```
elif combiner == "mean":
        embeddings = math_ops.segment_sum(embeddings, segment_ids)
        weight_sum = math_ops.segment_sum(weights, segment_ids)
        embeddings = math_ops.div(embeddings, weight_sum, name=name)
```

Should this be fixed with `math_ops. div_no_nan ` ?

### PR DESCRIPTION
**Pull request by ashutosh1919**
Fixes #36359. @Saduf2019 , @gowthamkpr and @yongtang, Please Review this one.

### CODE REVIEW

**Review by alextp — COMMENTED** [review]
Can you add a test which fails before this change? That will help prevent this from regressing.

Thanks for the contribution!

**Comment by ashutosh1919** [comment]
@alextp, Test case checking function is added. Please review PR.

**Comment by ashutosh1919** [comment]
@alextp, Test cases are added. Please review it

**Comment by ashutosh1919** [comment]
cc @mihaimaruseac

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by gbaned** [comment]
@ashutosh1919 Can you please check build failures? Thanks!

**Comment by ashutosh1919** [comment]
@gbaned & @mihaimaruseac , Resolved ci error

**Review by mihaimaruseac — DISMISSED** [review]

**Review by rthadur — DISMISSED** [review]

**Comment by mihaimaruseac** [comment]
Tests fail internally

```
    tf_embedding_sum = embedding_sum.numpy()
AttributeError: 'Tensor' object has no attribute 'numpy'
```

**Comment by ashutosh1919** [comment]
@mihaimaruseac , I think it gives error because in TF-2.x, eager execution is enabled by default. Is there any suggestion from your side to deal with this type of issue?

**Comment by mihaimaruseac** [comment]
All code should be eager first and working on 2.x behavior.

**Comment by ashutosh1919** [comment]
@mihaimaruseac , Please run ci, I hope it should work now.

**Review by mihaimaruseac — APPROVED** [review]

**Review by rthadur — APPROVED** [review]

**Comment by mihaimaruseac** [comment]
Still failing

```
    tf_embedding_sum = ops.convert_to_tensor(embedding_sum).numpy()
AttributeError: 'Tensor' object has no attribute 'numpy'
```

**Comment by ashutosh1919** [comment]
@mihaimaruseac , It is clearly mentioned [here](https://www.tensorflow.org/guide/eager#setup_and_basic_usage) that eager execution is compatible with `numpy()` function and can be applied on `tf.Tensor`. I am not able to catch the error. Can you please give me more info about the error message?

**Comment by mihaimaruseac** [comment]
It's probably an error caused by the internal integration. Let me dive into it and I will be back here with more details.

**Comment by ashutosh1919** [comment]
@mihaimaruseac , any updates on this PR?

**Comment by mihaimaruseac** [comment]
Unfortunately I didn't yet get time to fully fix it. I'll dedicate some more time tomorrow if nothing else pops up and preempts from this

**Comment by gbaned** [comment]
@mihaimaruseac Any update on this PR, please. Thanks!

**Comment by gbaned** [comment]
@mihaimaruseac Any update on this PR, please. Thanks!

**Comment by gbaned** [comment]
@mihaimaruseac Any update on this PR, please. Thanks!

**Review by mihaimaruseac — APPROVED** [review]

File: tensorflow/python/kernel_tests/embedding_ops_test.py [hunk 1]
```diff
@@ -748,6 +748,29 @@ def testEmbeddingLookupSparse(self):
...
+      sp_weights = sparse_tensor.SparseTensor(
+          constant_op.constant([[1, 0], [3, 0]], dtypes.int64),
+          constant_op.constant([1, 1], dtypes.float32),
+          constant_op.constant([4, 1], dtypes.int64))
+
+      for combiner in ["sum", "mean", "sqrtn"]:
+        embedding_sum = embedding_ops.embedding_lookup_sparse(
+            x, sp_ids, sp_weights, combiner=combiner)
+
+        tf_embedding_sum = ops.convert_to_tensor(embedding_sum).numpy()
```
Comment (rthadur) [inline tensorflow/python/kernel_tests/embedding_ops_test.py:768]: @ashutosh1919 can you please check this error 

`AttributeError: 'Tensor' object has no attribute 'numpy'
`

**Review by rthadur — CHANGES_REQUESTED** [review]

**Comment by gbaned** [comment]
@ashutosh1919  Any update on this PR? Please. Thanks!

**Comment by ashutosh1919** [comment]
@googlebot I fixed it.

**Comment by ashutosh1919** [comment]
@mihaimaruseac , @rthadur can you please approve it? Want to see if it passes the tests now

**Comment by ashutosh1919** [comment]
@mihaimaruseac , @rthadur - I have ran the bazel tests as well as sanity tests. Both are passing for the code which I changed. But the internal tests are failing here. I think the problem is with something else. Can you please look?

```
10,336 / 10,998] 8 actions running
Target //tensorflow/python/kernel_tests:embedding_ops_test up-to-date:
  bazel-bin/tensorflow/python/kernel_tests/embedding_ops_test
INFO: Elapsed time: 15494.530s, Critical Path: 2387.59s
INFO: 7668 processes: 213 internal, 7455 local.
INFO: Build completed successfully, 7668 total actions
//tensorflow/python/kernel_tests:embedding_ops_test                      PASSED in 67.4s
  Stats over 20 runs: max = 67.4s, min = 9.5s, avg = 26.4s, dev = 19.4s
```

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**